hazelcast/src/main/java/com/hazelcast/config/JoinConfig.java

```diff
@@ -118,16 +118,27 @@ public void verify() {
         if (getMulticastConfig().isEnabled() && getAwsConfig().isEnabled()) {
             throw new InvalidConfigurationException("Multicast and AWS join can't be enabled at the same time");
         }
+        verifyDiscoveryProviderConfig();
+    }
 
+    /**
+     * Verifies this JoinConfig is valid. When Discovery SPI enabled other discovery
+     * methods should be disabled
+     *
+     * @throws IllegalStateException when the join config is not valid.
```

Comment:
`@throws IllegalStateException when the join config is not valid.` <-- that is not correct.